packages/core/src/model/nodes/ChatNode.ts

```diff
@@ -910,37 +910,25 @@ export class ChatNodeImpl extends NodeImpl<ChatNode> {
             }
 
             if (functionCalls.length > 0) {
-              if (isMultiResponse) {
-                output['function-call' as PortId] = {
-                  type: 'object[]',
-                  value: functionCalls.map((functionCalls) => ({
-                    name: functionCalls[0]?.name,
-                    arguments: functionCalls[0]?.lastParsedArguments,
-                    id: functionCalls[0]?.id,
-                  })),
-                };
-              } else {
-                if (this.data.parallelFunctionCalling) {
-                  console.dir({ functionCalls });
-                  output['function-calls' as PortId] = {
-                    type: 'object[]',
-                    value: functionCalls[0]!.map((functionCall) => ({
-                      name: functionCall.name,
-                      arguments: functionCall.lastParsedArguments,
-                      id: functionCall.id,
-                    })),
-                  };
-                } else {
-                  output['function-call' as PortId] = {
-                    type: 'object',
-                    value: {
-                      name: functionCalls[0]![0]?.name,
-                      arguments: functionCalls[0]![0]?.lastParsedArguments,
-                      id: functionCalls[0]![0]?.id,
-                    } as Record<string, unknown>,
-                  };
-                }
-              }
+							if (isMultiResponse) {
+								output['function-call' as PortId] = {
+									type: 'object[]',
+									value: functionCalls.flat().map((functionCall) => ({
```

Comment:
Are you sure that `.flat()` is right here? If there are multiple LLM calls that all make multiple function calls, then the output should be `object[][]` - an array per LLM call

Reply:
IMHO that would have the same issue as above, changing expectations. I'll leave that up to you to decide and I can make the change

Reply:
You're right, but setting the LLM to return multiple responses _already_ arrayifies the string output, so it would make sense to arrayify the function call output as well. I consider it bug, and I doubt enough people are using multi-response with parallel function calling for it to be a big enough problem